### PR TITLE
Nc news 8

### PR DESCRIPTION
**Pull request by Deadline76**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -138,3 +137,23 @@ describe('/api/articles', () => {
         });
     })
 })
+describe('PATCH: /api/articles/:article_id', () => {
+    test.only('PATCH:200 returns an article object with the updated vote count', () => {
```
Comment (EddTheDuck) [inline __tests__/app.test.js]: Remove the .only please and resubmit!

**Review by EddTheDuck — CHANGES_REQUESTED** [review]
There is a .only in your test sweet so I immediately have to ping it back to you!

We need to make sure all your tests are passing before you submit a PR

File: __tests__/app.test.js [hunk 2]
```diff
@@ -138,3 +137,85 @@ describe('/api/articles', () => {
...
+    })
+    test('DELETE:404 responds with an appropriate error message when given a non-existent id', () => {
+      return request(app)
+        .delete('/api/comments/999')
+        .expect(404)
+        .then(({body}) => {
+          expect(body.msg).toBe('comment does not exist');
+      })
+    })
+    test('DELETE:404 responds with an appropriate error message when given an invalid id', () => {
```
Comment (dappernerddesigns) [inline __tests__/app.test.js:305]: Check your descriptions, you've written 404 here but asserting a 400.

File: controllers/controllers.js [hunk 3]
```diff
@@ -51,8 +56,40 @@ const getArticles = (req, res, next) => {
     })
 }
 
+const patchArticleVotes = (req, res, next) => {
+    const {article_id} = req.params
+    const {inc_votes} = req.body
 
+    if(!inc_votes || typeof inc_votes !== 'number' || Object.keys(req.body).length > 1){
+        return res.status(400).send({ msg: 'invalid post object'})
```
Comment (dappernerddesigns) [inline controllers/controllers.js]: This is not a Promise reject, keep your error handling in your model and you can then check your data structure and reject accordingly.

**Review by dappernerddesigns — APPROVED** [review]
I'd like you to consider the comments below before moving on. But really good work so far, looks like you hit your stride!

**Review by EddTheDuck — APPROVED** [review]
Looks good!

**Pull request merged**